vault/resource_mongodbatlas_secret_backend.go

```diff
@@ -0,0 +1,132 @@
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"golang.org/x/net/context"
+)
+
+var mongodbAtlasAPIFields = []string{
+	consts.FieldPublicKey,
+	consts.FieldPrivateKey,
+}
+
+func mongodbAtlasSecretBackendResource() *schema.Resource {
+	return provider.MustAddMountMigrationSchema(&schema.Resource{
+		CreateContext: mongodbAtlasSecretBackendCreateUpdate,
+		ReadContext:   ReadContextWrapper(mongodbAtlasSecretBackendRead),
+		UpdateContext: mongodbAtlasSecretBackendCreateUpdate,
+		DeleteContext: mongodbAtlasSecretBackendDelete,
+		CustomizeDiff: getMountCustomizeDiffFunc(consts.FieldPath),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldBackend: {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Path where MongoDB Atlas secret backend will be mounted",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			consts.FieldPath: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Path where MongoDB Atlas configuration is located",
+			},
+			consts.FieldPublicKey: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Public Programmatic API Key used to authenticate with the MongoDB Atlas API",
+			},
+			consts.FieldPrivateKey: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Private Programmatic API Key used to connect with MongoDB Atlas API",
+			},
+		},
+	})
+}
+
+func mongodbAtlasSecretBackendCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	backend := d.Get(consts.FieldBackend).(string)
+	privateKey := d.Get(consts.FieldPrivateKey).(string)
+	publicKey := d.Get(consts.FieldPublicKey).(string)
+
+	data := map[string]interface{}{
+		consts.FieldPrivateKey: privateKey,
+		consts.FieldPublicKey:  publicKey,
+	}
+
+	path := backend + "/config"
+	log.Printf("[DEBUG] Writing MongoDB Atlas config at %q", path)
+
+	_, err := client.Logical().Write(path, data)
+	if err != nil {
+		return diag.Errorf("error writing to %q, err=%s", path, err)
+	}
+
+	log.Printf("[DEBUG] Mounted MongoDB Atlas backend at %q", path)
+	d.SetId(path)
+
+	return mongodbAtlasSecretBackendRead(ctx, d, meta)
+}
+
+func mongodbAtlasSecretBackendRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	path := d.Id()
+	if err := d.Set(consts.FieldPath, path); err != nil {
+		return diag.FromErr(err)
+	}
+
+	backend := strings.Trim(path, "/config")
```

Comment:
Did some testing and realized this should be `TrimSuffix` instead 😅 
```suggestion
	backend := strings.TrimSuffix(path, "/config")
```